node/node.go

```diff
@@ -257,12 +275,16 @@ func New(host p2p.Host, consensusObj *consensus.Consensus, db ethdb.Database) *N
 
 		node.TxPool = core.NewTxPool(core.DefaultTxPoolConfig, params.TestChainConfig, chain)
 		node.Worker = worker.New(params.TestChainConfig, chain, node.Consensus, pki.GetAddressFromPublicKey(node.SelfPeer.PubKey), node.Consensus.ShardID)
-		if node.Role == BeaconLeader || node.Role == BeaconValidator {
-			node.CurrentStakes = make(map[common.Address]int64)
-		}
+
 		utils.GetLogInstance().Debug("Created Genesis Block", "blockHash", chain.GetBlockByNumber(0).Hash().Hex())
 		node.Consensus.ConsensusBlock = make(chan *consensus.BFTBlockInfo)
 		node.Consensus.VerifiedNewBlock = make(chan *types.Block)
+
+		// Setup one time smart contracts
+		node.AddFaucetContractToPendingTransactions()
```

Comment:
why did we move this logic in node_genesis and put it back here?

Reply:
This is not part of genesis. This is added as new transaction which goes into second block.